### ISSUE TITLE
Bug: Training does not appear to be occuring

### ISSUE THREAD

**Issue by FreezePhoenix** [tensorflow/tfjs#259]
#### TensorFlow.js version
0.10.0
#### Browser version
Chrome
#### Describe the problem or feature request

I'm trying to train it on vectors, with an input layer of 300 and an output layer of 300. It does not appear to be working, it always returns an array which corresponds the most with the vector for the word "in".

I may be using a linear model, if so please tell me how I would create a different model.

#### Code to reproduce the bug / link to feature request

https://codepen.io/Pheo/pen/BxRvKZ?editors=1001

**Comment by caisq** [comment]
@FreezePhoenix  I played around the code you provided on CodePen a little if I add the following line to the `then` block for `model.fit()`: 

```js
then(history =>{
        console.log("Finished!...")
        console.log(JSON.stringify(history.history))
        return;
      })
```

I can see that the loss value decreases over the three epochs. But the change was very small. Have you tried increasing the learning rate? If you do `this.model.compile({loss: "meanSquaredError", optimizer: 'sgd'});`, the learning rate you get from the default SGD is a relatively small value (0.1).

You can try using a larger learning rate, e.g.,

```js
const optimizer = tf.train.sgd(1);
      this.model.compile({loss: "meanSquaredError", optimizer: optimizer
```

You can also play around with
* Increasing number of training epochs
* Build a larger and more complex model

They way you are adding two `dense` layers without any activation function makes the two layers essentially one layer. You need to add nonlinear activations such as `relu`.

This issue doesn't seem to be any bug in tfjs, but detailed ML practice questions. So I'll close this issue now.
